c-cannons/bangazon-initial_site
# Paperclip

### Added Paperclip Gem

Paperclip is intended as an easy file attachment library for ActiveRecord. The intent behind it was to keep setup as easy as possible and to treat files as much like other attributes as possible. This means they aren't saved to their final locations on disk, nor are they deleted if set to nil, until ActiveRecord::Base#save is called. It manages validations based on size and presence, if required. It can transform its assigned image into thumbnails if needed, and the prerequisites are as simple as installing ImageMagick (which, for most modern Unix-based systems, is as easy as installing the right packages). Attached files are saved to the filesystem and referenced in the browser by an easily understandable specification, which has sensible and useful defaults.

#### Descriptions

Reference: [https://richonrails.com/articles/getting-started-with-paperclip](https://richonrails.com/articles/getting-started-with-paperclip)

Paperclip is already in our Gemfile, but you must install ImageMagick.

`brew install imagemagick`

Then `bundle install` again to give Paperclip access to ImageMagick

You may need to delete your cookies to end any previous sessions.

`rails db:migrate`

Sign Up/Login with a Customer

`/products/new` should now have an input field that allows you to upload a photo.

`/products/show` should display photo after clicking the `Sell` button

#### Steps to Test

Follow the above steps.

Create a customer and a product.

See if it throws an error or allows you to upload the file.

Check your database to see where the photo info is stored in the `Products` table

#### Link to Feature Ticket

https://github.com/c-cannons/bangazon-initial_site/issues/14
